packages/email-notification-dispatcher/src/handlers/eserviceTemplates/handleEserviceTemplateNameUpdated.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  EmailNotificationMessagePayload,
+  EService,
+  EServiceIdDescriptorId,
+  fromEServiceTemplateV2,
+  generateId,
+  missingKafkaMessageDataError,
+  NotificationType,
+  TenantId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import {
+  eventMailTemplateType,
+  retrieveHTMLTemplate,
+  retrieveLatestPublishedDescriptor,
+} from "../../services/utils.js";
+import { EserviceTemplateNameUpdatedHandlerParams } from "../handlerCommons.js";
+
+const notificationType: NotificationType = "templateStatusChangedToProducer";
+
+export async function handleEServiceTemplateNameUpdated(
+  params: EserviceTemplateNameUpdatedHandlerParams
+): Promise<EmailNotificationMessagePayload[]> {
+  const {
+    eserviceTemplateV2Msg,
+    oldName,
+    readModelService,
+    templateService,
+    userService,
+    correlationId,
+  } = params;
+
+  if (!eserviceTemplateV2Msg) {
+    throw missingKafkaMessageDataError(
+      "eserviceTemplate",
+      "EServiceTemplateNameUpdated"
+    );
+  }
+
+  const eserviceTemplate = fromEServiceTemplateV2(eserviceTemplateV2Msg);
+
+  const [htmlTemplate, eservices] = await Promise.all([
+    retrieveHTMLTemplate(
+      eventMailTemplateType.eserviceTemplateNameUpdatedMailTemplate
+    ),
+    readModelService.getEServicesByTemplateId(eserviceTemplate.id),
+  ]);
+
+  const instantiatorEserviceMap = Object.fromEntries(
+    eservices.reduce<Map<TenantId, EService[]>>((acc, eservice) => {
+      const current = acc.get(eservice.producerId) ?? [];
+      acc.set(eservice.producerId, [...current, eservice]);
+      return acc;
+    }, new Map())
+  );
+
+  const tenantIds: TenantId[] = Object.keys(instantiatorEserviceMap).map(
+    (tenantId) => unsafeBrandId(tenantId)
+  );
+
+  const instantiators = await readModelService.getTenantsById(tenantIds);
+
+  const tenantUsers =
```

Comment:
As discussed, this could be simplified using the modified `getRecipientsForTenant` in my PR #2446, but we can change this later, considering that #2446 won't be merged until we understand the new Selfcare integration strategy, so I can maybe add simplification of this part to my PR

The goal is to use `getRecipientsForTenant` also in these handlers, so that all handlers use the same function and that role checking can be centralized there

(Same for other handlers)